client/Windows/wf_channels.c

```diff
@@ -31,6 +31,72 @@
 #include <freerdp/log.h>
 #define TAG CLIENT_TAG("windows")
 
+static BOOL encomsp_toggle_control(EncomspClientContext* encomsp, BOOL control)
+{
+	ENCOMSP_CHANGE_PARTICIPANT_CONTROL_LEVEL_PDU pdu;
+
+	if (!encomsp)
+		return FALSE;
+
+	pdu.ParticipantId = 0;
+	pdu.Flags = ENCOMSP_REQUEST_VIEW;
+
+	if (control)
+		pdu.Flags |= ENCOMSP_REQUEST_INTERACT;
+
+	encomsp->ChangeParticipantControlLevel(encomsp, &pdu);
+	return TRUE;
+}
+
+static UINT
+wf_encomsp_participant_created(EncomspClientContext* context,
+                               const ENCOMSP_PARTICIPANT_CREATED_PDU* participantCreated)
+{
+	wfContext* wf;
+	rdpSettings* settings;
+	BOOL request;
+
+	if (!context || !context->custom || !participantCreated)
+		return ERROR_INVALID_PARAMETER;
+
+	wf = (wfContext*)context->custom;
+	WINPR_ASSERT(wf);
+
+	settings = wf->common.context.settings;
+
+	if (!settings)
```

Comment:
Shouldn't that test be a `WINPR_ASSERT` instead of a simple `if()` like if `settings == NULL` was an acceptable and sometime expected situation (when `settings` is NULL that means a code error on our side).